From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vesper Materializer <vesper-materializer@users.noreply.github.com>
Date: Fri, 21 Aug 2026 14:14:19 +0000
Subject: [PATCH 02/24] fix: adapt dynamic theme to inline media send

Signed-off-by: Vesper Materializer <vesper-materializer@users.noreply.github.com>
---
 .../securesms/mediasend/v3/MediaSendV3Activity.kt   | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/app/src/main/java/org/thoughtcrime/securesms/mediasend/v3/MediaSendV3Activity.kt b/app/src/main/java/org/thoughtcrime/securesms/mediasend/v3/MediaSendV3Activity.kt
index f59670f6d9..29a5393dd5 100644
--- a/app/src/main/java/org/thoughtcrime/securesms/mediasend/v3/MediaSendV3Activity.kt
+++ b/app/src/main/java/org/thoughtcrime/securesms/mediasend/v3/MediaSendV3Activity.kt
@@ -67,6 +67,8 @@ import org.thoughtcrime.securesms.safety.SafetyNumberBottomSheet
 import org.thoughtcrime.securesms.scribbles.StickerSelectActivityContract
 import org.thoughtcrime.securesms.util.CommunicationActions
 import org.thoughtcrime.securesms.util.DateUtils
+import org.thoughtcrime.securesms.util.DynamicNoActionBarTheme
+import org.thoughtcrime.securesms.util.DynamicTheme
 import org.signal.core.ui.R as CoreUiR
 
 /**
@@ -82,6 +84,8 @@ class MediaSendV3Activity :
   ScheduleMessageTimePickerBottomSheet.ScheduleCallback,
   ScheduleMessageDialogCallback {
 
+  private val theme: DynamicTheme = DynamicNoActionBarTheme()
+
   private val contractArgs: MediaSendFlowActivityContract.Args by lazy { MediaSendFlowActivityContract.Args.fromIntent(intent) }
 
   private val viewModel: MediaSendFlowViewModel by viewModels { MediaSendFlowViewModel.Factory(args = contractArgs) }
@@ -106,6 +110,15 @@ class MediaSendV3Activity :
     super.attachBaseContext(newBase)
   }
 
+  override fun onResume() {
+    super.onResume()
+    theme.onResume(this)
+  }
+
+  override fun onPreCreate() {
+    theme.onCreate(this)
+  }
+
   override fun onCreate(savedInstanceState: Bundle?, ready: Boolean) {
     if (resources.getWindowBreakpoint() !is WindowBreakpoint.Small) {
       requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
